test/aws/rails/middleware/elastic_beanstalk_worker_listener_test.rb

```diff
@@ -0,0 +1,95 @@
+require 'test_helper'
+require_relative 'elastic_beanstalk_job'
+require_relative 'elastic_beanstalk_periodic_task'
+
+module Aws
+  module Rails
+    describe ElasticBeanstalkWorkerListener do
+      # Simple mock Rack app that always returns 200
+      let(:mock_rack_app) { ->(_) { [200, { 'Content-Type' => 'text/plain' }, ['OK']] } }
+
+      let(:logger) { double(error: nil, debug: nil, warn: nil) }
+
+      it 'passes request through if user-agent is not SQS Daemon' do
+        mock_rack_env = create_mock_env('127.0.0.1', 'not-aws-sqsd')
+
+        test_middleware = ElasticBeanstalkWorkerListener.new(mock_rack_app)
+        response = test_middleware.call(mock_rack_env)
+
+        expect(response[0]).to eq(200)
+        expect(response[2]).to eq(['OK'])
+      end
+
+      it 'returns forbidden when called from untrusted source' do
+        mock_rack_env = create_mock_env('1.2.3.4', 'aws-sqsd/1.1')
+
+        test_middleware = ElasticBeanstalkWorkerListener.new(mock_rack_app)
+        response = test_middleware.call(mock_rack_env)
+
+        expect(response[0]).to eq(403)
+      end
+
+      it 'successfully invokes job when passed through request body' do
+        # Stub execute call to avoid invoking Active Job callbacks
+        allow(ActiveJob::Base).to receive(:execute).and_return(nil)
```

Comment:
can change this from allow to expect since this is specifically what we are testing for here.

Reply:
Makes sense - first time working with RSpec and still getting a sense of its capabilities. Will need to read up on their docs more.